File: Old Java Code/Client/org/hardgate/familytree/formatters/BirthOutputFormatter.java
/*
 * Created on 04-Jan-2005
 *
 */
package org.hardgate.familytree.formatters;

import java.io.PrintWriter;

import org.hardgate.familytree.registrations.BirthRegistration;
import org.hardgate.familytree.registrations.Registration;

/**
 * @author db
 *
 */
public class BirthOutputFormatter extends BaseOutputFormatter {

    /* (non-Javadoc)
     * @see org.hardgate.familytree.formatters.BaseOutputFormatter#printHeader(java.io.PrintWriter)
     */
    public void printHeader (PrintWriter output) {
        output.print("Surname,Forenames,Age,Occupation,Residence,");
        output.print("DateOfBirth,PlaceOfBirth,BestLocation,");
        output.print("FathersName,FathersOccupation,FatherDeceased,");
        output.print("MothersName,MothersOccupation,MotherDeceased,");
        output.print("Gender,ParentsMarriageDate,ParentsMarriagePlace");
        output.println();
    }

    /* (non-Javadoc)
     * @see org.hardgate.familytree.formatters.BaseOutputFormatter#printItem(org.hardgate.familytree.Registration, java.io.PrintWriter)
     */
    public void printItem (Registration reg, PrintWriter output) {
        BirthRegistration b = (BirthRegistration) reg;
        printFamilyGroup(b.getRegistrationDate(), b.getFamilyGroup(), output);
        output.print("\""); output.print(b.getGender()); output.print("\",");
        output.print("\""); output.print(b.getParentsMarriageDate()); output.print("\",");
        output.print("\""); output.print(b.getParentsMarriageLocation()); output.println("\"");
    }
}

File: FTAnalyser/Formatters/BirthOutputFormatter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using System.IO;

namespace FTAnalyzer
{
    public class BirthOutputFormatter : BaseOutputFormatter {

        public override void printHeader(TextWriter output)
        {
            output.Write("Surname,Forenames,Age,Occupation,Residence,");
            output.Write("DateOfBirth,PlaceOfBirth,BestLocation,");
            output.Write("FathersName,FathersOccupation,FatherDeceased,");
            output.Write("MothersName,MothersOccupation,MotherDeceased,");
            output.Write("Gender,ParentsMarriageDate,ParentsMarriagePlace");
            output.WriteLine();
        }

        public override void printItem(Registration reg, TextWriter output)
        {
            BirthRegistration b = (BirthRegistration) reg;
            printFamilyGroup(b.RegistrationDate, b.FamilyGroup, output);
            output.Write("\""); output.Write(b.Gender); output.Write("\",");
            output.Write("\""); output.Write(b.ParentsMarriageDate); output.Write("\",");
            output.Write("\""); output.Write(b.ParentsMarriageLocation); output.WriteLine("\"");
        }
    }
}